### PR TITLE
Heroku

### PR DESCRIPTION
**Pull request by jonnyarnold**
We are now deployed on Heroku. If we have succeeded, Samuel should post the guidelines on this Pull Request.

Here goes...

### CODE REVIEW

File: lib/samuel/api/server.ex [hunk 1]
```diff
@@ -12,7 +12,7 @@ defmodule Samuel.API.Server do
   A shell script that starts the server in production mode can be found in
   `bin/prod_run.sh`
   """
-  def start do
+  def start(port \\ 4000) do
```
Comment (lpil) [inline lib/samuel/api/server.ex]: ? What happened to 12 factor?

File: lib/tasks/serve.ex [hunk 2]
```diff
@@ -0,0 +1,8 @@
+defmodule Mix.Tasks.Serve do
+  use Mix.Task
+
+  def run([port_str]) do
```
Comment (lpil) [inline lib/tasks/serve.ex]: Why are you rebuilding what we already have?

**Comment by lpil** [comment]
https://github.com/reevoo/samuel/blob/master/bin/prod_run.sh

**Comment by lpil** [comment]
:(

**Comment by jonnyarnold** [comment]
Well that didn't work.

```
[error] Ranch listener Samuel.API.HTTP had connection process started with :cowboy_protocol:start_link/4 at #PID<0.228.0> exit with reason: {{:noproc, {:gen_server, :call, 
...[truncated]...
```

**Comment by lpil** [comment]
I think this starts in development env?

Where did that message come from?

File: lib/tasks/serve.ex (hunk 2 above)
Comment (jonnyarnold) [inline lib/tasks/serve.ex]: Derp, didn't see the bin.

So the procfile has the port pushed in (using `$PORT`). I assume we need to push this through to the server? (It's not an ENV...)

Comment (lpil) [inline lib/tasks/serve.ex]: That's silly, Heroku are all about env. Why is port different? Are we sure it's not both in $PORT, and an env variable?

Failing that...

```bash
PORT=$PORT bin/whatever.sh
```

???

**Pull request closed without merging**